### ISSUE TITLE
Python 2 MultiServer Mode, ValueError: [ERROR]: Failed to connect address

### ISSUE THREAD

**Issue by julio16101** [abhiTronix/vidgear#57]
I use the testing branch of vidgear to use use Multiserver Mode.

running this part of code on a raspberry pi 3B + that transmits the video from a USB camera to a computer.

```
from vidgear.gears import NetGear
from vidgear.gears import CamGear
import cv2

#Open live video stream from device at index 0
stream = cv2.VideoCapture(0) 

#activate multiserver_mode
options = {'multiserver_mode': True, 'flag' : 0, 'copy' : False, 'track' : False}

#change following IP address '192.168.1.xxx' with Client's IP address and assign unique port address(for e.g 5566).
server = NetGear(address = 192.168.x.x, port = '5566', protocol = 'tcp',  pattern = 2, receive_mode = False, **options) # and keep rest of settings similar to Client

# infinite loop until [Ctrl+C] is pressed
while True:
	try: 
		(grabbed, frame) = stream.read()
		# read frames

		# check if frame is not grabbed
		if not grabbed:
			#if True break the infinite loop
			break

		# do something with frame and data(to be sent) here

		text = "Hello, I'm Server-1 at Port Address: 5566."

		# send frame and data through server
		server.send(frame, message = text)
	
	except KeyboardInterrupt:
		#break the infinite loop
		break

# safely close video stream.
stream.release()
# safely close server-1
server.close()
```
I get this error:
ValueError: [ERROR]: Failed to connect address: tcp://192.168.0.107:5566 and pattern: 2! Kindly recheck all parameters.

Tracking the problem, it originates in line 477:
`self.msg_socket.bind(protocol+'://' + str(address) + ':' + str(port))`

testing this code in python3 works fine.

Is the problem due to the python version?

**Comment by abhiTronix** [comment]
@julio16101 **Kindly paste full error log from your terminal(both server and client's end). Secondly, post the output of `hostname -I` command to check both Server's and Client's IP addresses.**

**Comment by julio16101** [comment]
The error message I receive on my raspberry pi, using python 2 is:

Unable to stop the stream: Invalid argument
[LOG]: Enabling Multi-Server Mode at PORT: 5566 on this device!
Traceback (most recent call last):
  File "cliente.py", line 13, in <module>
    **options)  # and keep rest of settings similar to Client
  File "/home/pi/facialix/gears/netgear.py", line 477, in __init__
    raise ValueError('[ERROR]: Failed to connect address: {} and pattern: {}! Kindly recheck all parameters.'.format((protocol+'://' + str(address) + ':' + str(port)), pattern))
ValueError: [ERROR]: Failed to connect address: tcp://192.168.0.107:5566 and pattern: 2! Kindly recheck all parameters.

I don't receive any errors on my computer

I have previously used zmq for sending video from a raspberry with python 2 to my machine with python 3 without any problem.

The IP of my computer that receives video is: 192.168.0.107
The ip of the raspberry pi that sends video is: 192.168.0.112
both are static
pi@raspberrypi:~/fx $ hostname -I
192.168.0.112


In addition to that I am using the test branch

**Comment by abhiTronix** [comment]
@julio16101 Why are you using `multi-server mode` when you are using only two machines i.e a raspberry pi and a computer? Secondly, when using `multi-server mode`, the port must be a **list of port addresses** of each unique server at the client's end and not a string. 

### Solution:

**Kindly make `port` address value to `list`, i.e use `['5566']` instead of `'5566'` in your code OR Refrain from using multi-server mode.**

**Comment by abhiTronix** [comment]
@julio16101 It's working fine at my machine, even with python 2.7. Kindly recheck your code and package version of each dependency, something is definitely wrong. 

**Also, anyways I'm removing support for python 2.7 legacies in the next release, therefore kindly refrain from using `vidgear v0.1.6 (including dev)` with python 2 and its legacies and switch over to python 3+ legacies as soon as possible. Goodluck**

**Comment by julio16101** [comment]
I have already updated the version of pyzmq and the connection port parameter.

I continue with the same error, I will try updating the version of python 2 to python 3, although as I depend on opencv, the installation is more complicated

**Comment by abhiTronix** [comment]
> I continue with the same error

@julio16101 Is your python 2 version is python 2.6 or greater? Kindly paste the output of `python -c 'import platform; print(platform.python_version())'` command from your terminal.

> I will try updating the version of python 2 to python 3, although as I depend on opencv, the installation is more complicated

`sudo pip2 install -U opencv-contrib-python` will do the trick. It's not always necessary/recommended to build OpenCV from scratch until unless you want the best performance.

**Comment by abhiTronix** [comment]
@julio16101 Also, What OS you're using on your computer? Windows or Linux?

**Comment by julio16101** [comment]
As I mentioned earlier, the problem I mention is in my raspberry pi 3B +.

I am using Raspbian as an operating system (Linux).

My Python version is 2.7.13

On my computer where I receive the video, I am using a version of ubuntu 16.04 LTS although in it I have no problem.

**Comment by abhiTronix** [comment]
@julio16101 Can you try using the server's IP address(i.e your raspberry pi's IP address) on both machines instead?

**Comment by julio16101** [comment]
@abhiTronix  You mean if I can communicate with raspberry?
If it is, it is correct, the communication exists with the IPs, besides they are static to avoid inconveniences.

Previously I had already tried sending video using sockets.

**Comment by abhiTronix** [comment]
> @abhiTronix You mean if I can communicate with raspberry?
> If it is, it is correct, the communication exists with the IPs, besides they are static to avoid inconveniences.
> 
> Previously I had already tried sending video using sockets.

@julio16101 No, I meant to use 

```python
# put address value to your raspberry's IP address.
server = NetGear(address = 192.168.x.x, port = '5566', protocol = 'tcp',  pattern = 2, receive_mode = False, **options) # and keep rest of settings similar to Client
```
in your code.

**Comment by julio16101** [comment]
@abhiTronix  The server (Raspberry) must point to the client's IP (PC), is that correct?

**Comment by abhiTronix** [comment]
@abhiTronix No, I meant,  both the addresses must point to server's (Raspberry PI ) IP address only.

**Comment by julio16101** [comment]
I just installed opencv in Python3 to perform the test.

It worked correctly with the minimum example, some problem should exist with some modules in python2.

**Comment by abhiTronix** [comment]
@julio16101 Did you tested the multiserver mode?  Is it working correctly?

**Comment by julio16101** [comment]
@abhiTronix  Multi server mode works correctly.
Something makes me think there was a problem in Python, likewise I will update all my devices to Python3

**Comment by abhiTronix** [comment]
> I Use PUB/SUB pattern
> The error message I receive on my raspberry pi, using python 2 is:
> 
> Unable to stop the stream: Invalid argument
> [LOG]: Enabling Multi-Server Mode at PORT: 5566 on this device!

**@julio16101 This is also a bug with NetGear API due to its broken implementation. This also been fixed in the next PR. Thanks for reporting.**

**Comment by abhiTronix** [comment]
**Successfully resolved and merged in commit: e918d13**
